ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -697,7 +697,20 @@ def bind_ovs_ports(self, br_name, dut_iface, injected_iface, vm_iface, disconnec
             VMTopology.cmd("ovs-ofctl add-flow %s table=0,in_port=%s,action=output:%s" % (br_name, dut_iface_id, injected_iface_id))
         else:
             # Add flow from external iface to a VM and a ptf container
-            VMTopology.cmd("ovs-ofctl add-flow %s table=0,in_port=%s,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            # Allow BGP, ICMP, SNMP packets and layer2 packets from DUT to neighbors
+            # Block other traffic from DUT to EOS for EOS's stability,
+            # Allow all traffic from DUT to PTF.
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=10,tcp,in_port=%s,tp_src=179,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=10,tcp,in_port=%s,tp_dst=179,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=10,tcp6,in_port=%s,tp_src=179,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
+            VMTopology.cmd("ovs-ofctl add-flow %s table=0,priority=10,tcp6,in_port=%s,tp_dst=179,action=output:%s,%s" % (br_name, dut_iface_id, vm_iface_id, injected_iface_id))
```

Comment:
Do we really need to listen to the BGP, SNMP packet in ptf?

Reply:
For BGP, that's because of test_bgpmon test case. It adds a bgp monitor on PTF and verify that DUT is attempting to establish connection to it, that's why I allow BGP flow into PTF.
For SNMP, test case test_snmp_loopback sends snmp packets between neighbors and DUT. I keep it for PTF just in case of any scenario that SNMP packets required between PTF and DUT in the future.